http_requests_base.py

```diff
@@ -131,9 +131,8 @@ def _process_response(self, response, signal):
 
             # otherwise, no dice on parsing the response body
             else:
-                self.logger.warning("Response body could not be parsed into "
+                raise ValueError("Response body could not be parsed into "
```

Comment:
I kept the warning message here but I could remove it since the ValueError exception block will put out a warning if needed